### PR TITLE
Add an option to override the default build container

### PR DESCRIPTION
**Pull request by mattytrentini**
Could even remove the `idf` option?

### CODE REVIEW

File: src/mpbuild/build.py [hunk 1]
```diff
@@ -42,15 +35,16 @@ def build_board(
...
 
-    if port == "esp32":
+    if port == "esp32" and not build_container_override:
         if not idf:
             idf = IDF_DEFAULT
         build_container += f":{idf}"
 
-    if variant:
-        extra_args.insert(0, f"BOARD_VARIANT={variant}")
+    variant = f"BOARD_VARIANT={variant}" if variant else ""
```
Comment (mattytrentini) [inline src/mpbuild/build.py:47]: I prefer variants specified as their own thing

File: src/mpbuild/build.py [hunk 2]
```diff
@@ -67,7 +61,7 @@ def build_board(
         f"{build_container} "
         f'bash -c "'
         f"git config --global --add safe.directory '*' 2> /dev/null;"
-        f'make -C mpy-cross && make -C ports/{port} submodules all BOARD={board} {args}"'
+        f'make -C mpy-cross && make -C ports/{port} submodules all BOARD={board} {variant} {args}"'
```
Comment (mattytrentini) [inline src/mpbuild/build.py:64]: I don't like that if `variant` and `args` are not specified we get an additional two spaces - but the workarounds are annoying!

Comment (andrewleech) [reply src/mpbuild/build.py:64]: Just try `variant = f"BOARD_VARIANT={variant}" if variant else r"\b"` ;-)

File: src/mpbuild/build.py [hunk 3]
```diff
@@ -2,37 +2,30 @@
...
     variant: Optional[str] = None,
     extra_args: Optional[List[str]] = [],
+    build_container_override: Optional[str] = None,
     idf: Optional[str] = None,
 ) -> None:
-    if port not in _build_containers.keys():
+    if port not in BUILD_CONTAINERS.keys():
         print(f"Sorry, builds are not supported for the {port} port at this time")
         # TODO(mst) Should raise an exception and abort with an error code
         return
```
Comment (andrewleech) [inline src/mpbuild/build.py:31]: FYI to raise exception to abort with an error code you can simply use:
``` python
raise SystemExit()
```

File: src/mpbuild/build.py (hunk 2 above)
Comment (mattytrentini) [reply src/mpbuild/build.py:64]: You know, I had the same silly thought too 😝

**Comment by andrewleech** [comment]
> Could even remove the `idf` option?

Maybe allow the user to just pass in just the tag or the whole container?
``` python
if build_container_override.startswith(":"):
    build_container = build_container.split(":")[0] + build_container_override
else:
    build_container = build_container_override
```

File: src/mpbuild/build.py (hunk 2 above)
Comment (mattytrentini) [reply src/mpbuild/build.py]: OK, the fix wasn't too ugly...

**Comment by mattytrentini** [comment]
> > Could even remove the `idf` option?
> 
> Maybe allow the user to just pass in just the tag or the whole container?

I can't decide if I prefer this or not. I think it impacts usability a little - and feels a little opaque? But it sure seems convenient...I'll kick the can down the road and raise an issue to capture the idea.

File: src/mpbuild/build.py (hunk 3 above)
Comment (mattytrentini) [reply src/mpbuild/build.py]: Updated

**Pull request merged**